app/src/main/java/com/example/keniphhomemaintenance/maintenance_screen/MaintenanceView.kt

```diff
@@ -0,0 +1,252 @@
+import com.example.keniphhomemaintenance.R
+
+val maintenanceViewModel = MaintenanceViewModel()
+
+@Composable
+fun MaintenanceScreen() {
+    val maintenanceList = maintenanceViewModel.maintenanceItemList.value
+    val isDialogOpen = remember { mutableStateOf(false) }
+
+    Scaffold(
+        floatingActionButton = {
+            FloatingActionButton(
+                modifier = Modifier.size(dimensionResource(id = R.dimen.add_maintenance_item_button_size)),
+                shape = CircleShape,
+                onClick = {
+                    isDialogOpen.value = true
+                }
+            ) {
+                Icon(Icons.Default.Add, contentDescription = "Button to add a maintenance item")
+            }
+        },
+        floatingActionButtonPosition = FabPosition.Center
+    ){
+            ShowDialog(isDialogOpen)
+    }
+
+    MaintenanceItemsList(maintenanceList)
+
+}
+
+
+
+@Composable
+fun MaintenanceCard(maintenanceItem: MaintenanceItem) {
+    Spacer(modifier = Modifier.width(dimensionResource(R.dimen.maintenance_card_start_padding)))
+    Row(
+        modifier = Modifier
+            .padding(vertical = dimensionResource(R.dimen.maintenance_card_row_vertical_padding))
+            .padding(start = dimensionResource(R.dimen.maintenance_card_row_start_padding))
+            .fillMaxWidth(),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Image(
+            painter = painterResource(R.drawable.ic_baseline_build_24), 
+            contentDescription = "Maintenance Item Type",
+            modifier = Modifier
+                .size(dimensionResource(R.dimen.maintenance_image_size))
+                .clip(CircleShape)
+        )
+        
+        Spacer(modifier = Modifier.width(dimensionResource(R.dimen.maintenance_image_end_padding)))
+
+        Column {
+            Text(
+                text = maintenanceItem.title,
+                color = MaterialTheme.colors.primaryVariant,
+                style = MaterialTheme.typography.h5
+            )
+            Spacer(modifier = Modifier.height(dimensionResource(R.dimen.maintenance_title_and_due_date_padding)))
+            Text(
+                text = maintenanceItem.dueDate,
+                color = Color.Black,
+                style = MaterialTheme.typography.body2
+            )
+            Text(
+                text = maintenanceItem.location,
+                color = Color.Black,
+                style = MaterialTheme.typography.body2
+
+            )
+        }
+    }
+}
+
+@Composable
+fun MaintenanceItemsList(maintenanceItems: List<MaintenanceItem>) {
+    LazyColumn(
+        modifier = Modifier
+            .fillMaxHeight()
+            .padding(bottom = dimensionResource(R.dimen.maintenance_item_list_bottom_padding))
+    ) {
+        items(maintenanceItems) { maintenanceItems ->
+            MaintenanceCard(maintenanceItems)
+        }
+    }
+}
+
+@Composable
+fun ShowDialog(isDialogOpen: MutableState<Boolean>) {
```

Comment:
functions are verbs, and `ShowDialog` is a verb, but I'm confused in jetpack compose which one we are supposed to use since a `@Composable` is a noun (layout). Is this better off being named ShowDialog (verb) or something like NewMaintenanceItemDialog (noun)

Reply:
For composables, since they just provide a UI asset, I would prefer we use noun clauses rather than verb clauses. They're technically functions but don't act like them.